OpenRA.Mods.Common/Activities/UnloadCargo.cs

```diff
@@ -121,10 +121,10 @@ public override bool Tick(Actor self)
 					var move = actor.Trait<IMove>();
 					var pos = actor.Trait<IPositionable>();
 
-					pos.SetPosition(self, exitSubCell.Value.First, exitSubCell.Value.Second);
+					pos.SetPosition(actor, exitSubCell.Value.First, exitSubCell.Value.Second);
```

Comment:
I haven't actually tested if this makes a difference (it should, I didn't spot anything on a quick glance though). It only seems logical to me however, that the `IPositionable` trait of `actor` should actually change the state of `actor` and not `self`.

Reply:
Ouch! This is probably the actual cause of the glitch. (And if not, it would definitely be causing some other glitch at some point).

Reply:
Changing this does not change what I see in #17009 fwiw.